api/src/Doctrine/GetHeistExtension.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Doctrine;
+
+use ApiPlatform\Doctrine\Orm\Extension\QueryCollectionExtensionInterface;
+use ApiPlatform\Doctrine\Orm\Extension\QueryItemExtensionInterface;
+use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Metadata\Operation;
+use App\Entity\Heist;
+use App\Entity\User;
+use App\Enum\HeistVisibilityEnum;
+use Doctrine\ORM\QueryBuilder;
+use Symfony\Bundle\SecurityBundle\Security;
+
+class GetHeistExtension implements QueryCollectionExtensionInterface, QueryItemExtensionInterface
+{
+    public function __construct(private readonly Security $security)
+    {
+    }
+
+    /**
+     * @param array<string> $context
+     */
+    public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, Operation $operation = null, array $context = []): void
+    {
+        $this->addWhere($queryBuilder, $resourceClass);
+    }
+
+    /**
+     * @param array<string> $context
+     * @param array<mixed>  $identifiers
+     */
+    public function applyToItem(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, array $identifiers, Operation $operation = null, array $context = []): void
+    {
+        $this->addWhere($queryBuilder, $resourceClass);
+    }
+
+    private function addWhere(QueryBuilder $queryBuilder, string $resourceClass): void
+    {
+        /* @var User */
+        if (Heist::class !== $resourceClass) {
+            return;
+        }
+
+        /* @var User */
+        $user = $this->security->getUser();
+
+        if (null === $user || !array_intersect($user->getRoles(), [User::ROLE_ADMIN, User::ROLE_CONTRACTOR])) {
```

Comment:
```suggestion
        if (null === $user || !array_intersect($user->getRoles(), [User::ROLE_ADMIN, User::ROLE_CONTRACTOR])) {
```

Il faut utiliser `$this->security->isGranted`